### DISCUSSION TITLE
Platform field of vcpkg.json manifest does not recognize "freebsd"

### DISCUSSION THREAD

**Discussion by vserdyuk** [microsoft/vcpkg#21964] [Ideas]
**Describe the bug**
Platform field of vcpkg.json manifest does not recognize "freebsd"

**Environment**
- OS: FreeBSD 13
- Compiler: FreeBSD clang version 11.0.1 (git@github.com:llvm/llvm-project.git llvmorg-11.0.1-0-g43ff75f2c3fe)

**To Reproduce**
1. create vcpkg.json with the following contents:
```json
{
  "name": "myproj",
  "version": "0.0.1",
  "dependencies": [
    {
      "name": "gtest", "platform": "windows | (linux & !arm) | osx | freebsd"
    }
  ]
}
```
2. ./vcpkg install --triplet=x64-freebsd
3. See error
`Error: Unrecognized identifer name freebsd. Add to override list in triplet file.`
Library not installed

**Expected behavior**
No errors, libraries should be installed

**Failure logs** 
- see above

**Additional context**
The reason is that `src\vcpkg\platform-expression.cpp` has a hard-coded list of platform identifiers. An easy solution is to just add there a `freebsd` variant. Document that when adding a new community triplet, a contributor is encouraged to add a new entry there.
But maybe just allow all lowercase variants of `CMAKE_SYSTEM_NAME` used in triplets.

**Comment by PhoebeHui** [comment]
See https://github.com/microsoft/vcpkg-tool/pull/309

**Comment by autoantwort** [comment]
Can this be closed somehow?

**Comment by vserdyuk** [comment]
Fixed in https://github.com/microsoft/vcpkg-tool/pull/309
